From d08826c4fde512a79ca1322e322e81d891a7cd4a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger@applied-asynchrony.com>
Date: Tue, 23 Jul 2024 13:36:41 +0200
Subject: [PATCH] Fix building freestanding test with clang & enabled stack
 protector

The freestanding test is special as it requires reimplementation of
runtime/toolchain functionality. When building with clang where the
compiler is configured to enable the stack-protector by default,
building fails due to missing __stack_chk_fail symbols.
Thefore disable stack protection specifically for this test.

Additionally, disable the freestanding test entirely as it segfaults
at runtime.
---
 build/meson/meson/tests/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/meson/meson/tests/meson.build b/build/meson/meson/tests/meson.build
index 4cf86a635..c64b4f13c 100644
--- a/build/meson/meson/tests/meson.build
+++ b/build/meson/meson/tests/meson.build
@@ -47,7 +47,7 @@ test_exes = {
   },
   'freestanding': {
     'sources': files(lz4_source_root / 'tests/freestanding.c'),
-    'c_args': ['-ffreestanding', '-Wno-unused-parameter', '-Wno-declaration-after-statement'],
+    'c_args': ['-ffreestanding', '-fno-stack-protector', '-Wno-unused-parameter', '-Wno-declaration-after-statement'],
     'link_args': ['-nostdlib'],
-    'build': cc.get_id() in ['gcc', 'clang'] and
+    'build': false and cc.get_id() in ['gcc', 'clang'] and
       host_machine.system() == 'linux' and host_machine.cpu_family() == 'x86_64',
